sdk/log/crate/src/logger.rs

```diff
@@ -296,6 +317,12 @@ macro_rules! impl_log_for_unsigned_integer {
                         }
 
                         // There might not have been space for all the value.
+                        // JC: since this function uses an intermediate array
+                        // of `MaybeUninit<u8>`, it could be nicer to have this
+                        // function return the `MaybeUninit` array, so that the
+                        // logger decides how to deal with overflow if the
+                        // returned amount is too big. However, that isn't possible
+                        // at compile-time, so maybe this is the best option.
```

Comment:
Yeah, another option would be to change the return type for `write_with_args` to indicate the overflow. But since there is a truncate args, perhaps the truncation should be down to the each individual value?

Reply:
Hm... not sure, we can just stick with this I think